src/atomEffect.ts

```diff
@@ -49,18 +66,14 @@ export function atomEffect(
       }
       ++ref.inProgress
       try {
+        const setter = makeSetter(setSelf as Setter, ref) as Setter
         await ref.cleanup?.()
-        ref.cleanup = await effectFn(get, setSelf as Setter)
+        ref.cleanup = await effectFn(get, setter)
       } finally {
         --ref.inProgress
       }
     },
-    (
-      _get,
-      set,
-      a: WritableAtom<unknown, unknown[], unknown>,
-      ...args: unknown[]
-    ) => set(a, ...args)
+    (_get, set, ...args: Parameters<Setter>) => set(...args)
```

Comment:
Nice > Parameters.

Here's another coding idea instead of `makeSetter`:
```js
(get, set, ...args: Parameters<Setter>) => {
  const ref = get(refAtom)
  ++ref.inProgress
  try {
    return set(...args)
  } finally {
    --ref.inProgress
  }
}
```

We can eliminate `State` type too.

Reply:
hm, maybe I should make it a PR.

Reply:
I understand. That is very nice, thank you.

Reply:
#5